store/orm/orm.go

```diff
@@ -369,12 +364,30 @@ func (orm *ORM) CreateServiceAgreement(sa *models.ServiceAgreement) error {
 	return orm.DB.Create(sa).Error
 }
 
-// UnscopedJobRunsWithStatus returns all JobRuns, including ones that have been
-// soft deleted, which have the passed statuses.
-func (orm *ORM) UnscopedJobRunsWithStatus(statuses ...models.RunStatus) ([]models.JobRun, error) {
-	runs := []models.JobRun{}
-	err := orm.Unscoped().preloadJobRuns().Where("status IN (?)", statuses).Find(&runs).Error
-	return runs, err
+// UnscopedJobRunsWithStatus passes all JobRuns to a callback, one by one,
+// including those that were soft deleted.
+func (orm *ORM) UnscopedJobRunsWithStatus(cb func(*models.JobRun), statuses ...models.RunStatus) error {
+	var runIDs []string
+	err := orm.DB.Unscoped().
+		Table("job_runs").
+		Where("status IN (?)", statuses).
+		Order("created_at asc").
+		Pluck("ID", &runIDs).Error
+	if err != nil {
+		return fmt.Errorf("error finding job ids %v", err)
+	}
+
+	for _, id := range runIDs {
+		var run models.JobRun
+		err := orm.Unscoped().preloadJobRuns().First(&run, "id = ?", id).Error
```

Comment:
Wouldn't even a small batch of 10 here still have a dramatic 10x improvement?

Reply:
Nah they'd just need > 999 job runs to hit the limit, then your node is bricked.

Reply:
Maybe,  but my preference would still be to do that separately and to measure the difference.